mono_repo/lib/src/commands/init.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:path/path.dart' as p;
+
+const _pubspecFileName = 'pubspec.yaml';
+const _pkgCfgFileName = 'mono_pkg.yaml';
+const _repoCfgFileName = 'mono_repo.yaml';
+const _recursiveScanFlag = 'recursive';
+
+const _commentText =
+    '# See with https://github.com/dart-lang/mono_repo for details on this file\n';
+
+class InitCommand extends Command<void> {
+  @override
+  String get name => 'init';
+
+  @override
+  String get description => 'Scaffold a new mono repo.';
+
+  @override
+  void run() => scaffold(p.current, globalResults[_recursiveScanFlag] as bool);
+}
+
+void scaffold(String rootDir, bool recursive) {
+  configureDirectory(rootDir, recursive: recursive);
+}
+
+void configureDirectory(String rootDir,
+    {String currentDir, bool recursive = false}) {
+  currentDir ??= rootDir;
+
+  if (currentDir == rootDir) {
+    final repoCfgPath = p.join(rootDir, _repoCfgFileName);
+    if (!File(repoCfgPath).existsSync()) {
+      File(repoCfgPath).writeAsStringSync(_commentText);
+      print('Added $_repoCfgFileName to $rootDir');
+    } else {
+      print('$_repoCfgFileName already present in $rootDir. Exiting...');
+      return;
+    }
+  } else {
+    final pkgCfgPath = p.join(currentDir, _pkgCfgFileName);
+    final pubspecPath = p.join(currentDir, _pubspecFileName);
+
+    if (!File(pubspecPath).existsSync()) {
+      final pubspecContents = 'name: ${p.basename(currentDir)}\n';
+      File(pubspecPath).writeAsStringSync(pubspecContents);
+      print('Added $_pubspecFileName to $currentDir');
+    }
+
+    if (!File(pkgCfgPath).existsSync()) {
+      File(pkgCfgPath).writeAsStringSync(_commentText);
+      print('Added $_pkgCfgFileName to $currentDir');
+    }
+  }
+
+  final subdirs =
+      Directory(currentDir).listSync().whereType<Directory>().toList();
+  for (var subdir in subdirs) {
```

Comment:
I am a bit on the fence about this behavior - especially in the case of pre-existing packages we wouldn't want to go creating a whole bunch of pubspecs in all nested packages.

I am somewhat inclined to just not support this for the `init` command at all - or we could remove the behavior that creates a new package in all directories where a pubspec did not exist before.

Reply:
If we add the mono_pkg.yaml only to directories that already have a pubspec.yaml, we're assuming the user is never starting from scratch.

And I was initially adding the mono_pkg.yaml file only to the immediate subdirectories of root. But directory structures such as root/pkgs/package1,package2 are common too, so doing it recursively seemed to make more sense.

I guess you decide the behavior, and I'll update the code accordingly.

Reply:
I think it would be safest to add the mono_pkg.yaml only to directories that have a pubspec.yaml.